File: ms-java/src/main/java/com/jakubeeee/playground/msjava/organization/internal/DepartmentInitialSaver.java
package com.jakubeeee.playground.msjava.organization.internal;

import com.jakubeeee.playground.common.jpa.InitialSaver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.stream.Stream;

@Component
public class DepartmentInitialSaver extends InitialSaver<DepartmentEntity, DepartmentRepository> {

    @Autowired
    public DepartmentInitialSaver(DepartmentRepository repository) {
        super(repository);
    }

    @Override
    protected Stream<DepartmentEntity> initialEntities() {
        return Stream.of(
                new DepartmentEntity("Information_Technologies"),
                new DepartmentEntity("Marketing"),
                new DepartmentEntity("Human_Resources"),
                new DepartmentEntity("Research_And_Development"),
                new DepartmentEntity("Accounting")
        );
    }

}

File: ms-kotlin/src/main/kotlin/com/jakubeeee/playground/mskotlin/organization/internal/DepartmentInitialSaver.kt
package com.jakubeeee.playground.mskotlin.organization.internal

import com.jakubeeee.playground.common.jpa.InitialSaver
import org.springframework.stereotype.Component
import java.util.stream.Stream

@Component
open class DepartmentInitialSaver(repository: DepartmentRepository) : InitialSaver<DepartmentEntity, DepartmentRepository>(repository) {

    override fun initialEntities(): Stream<DepartmentEntity> {
        return Stream.of(
                DepartmentEntity("Information_Technologies"),
                DepartmentEntity("Marketing"),
                DepartmentEntity("Human_Resources"),
                DepartmentEntity("Research_And_Development"),
                DepartmentEntity("Accounting")
        )
    }

}